scrapy/pipelines/images.py

```diff
@@ -128,15 +131,33 @@ def get_images(self, response, request, info, *, item=None):
             raise ImageException("Image too small (%dx%d < %dx%d)" %
                                  (width, height, self.min_width, self.min_height))
 
-        image, buf = self.convert_image(orig_image)
+        if self._deprecated_convert_image is None:
+            self._deprecated_convert_image = 'response_body' not in get_func_args(self.convert_image)
+            if self._deprecated_convert_image:
+                warnings.warn('ImagesPipeline.convert_image() method overriden in a deprecated way, '
+                              'overriden method does not accept response_body argument.',
+                              category=ScrapyDeprecationWarning, stacklevel=1)
+
+        if self._deprecated_convert_image:
+            image, buf = self.convert_image(orig_image)
+        else:
+            image, buf = self.convert_image(orig_image, response_body=BytesIO(response.body))
         yield path, image, buf
 
         for thumb_id, size in self.thumbs.items():
             thumb_path = self.thumb_path(request, thumb_id, response=response, info=info)
-            thumb_image, thumb_buf = self.convert_image(image, size)
+            if self._deprecated_convert_image:
+                thumb_image, thumb_buf = self.convert_image(image, size)
+            else:
+                thumb_image, thumb_buf = self.convert_image(image, size, buf)
             yield thumb_path, thumb_image, thumb_buf
 
-    def convert_image(self, image, size=None):
+    def convert_image(self, image, size=None, response_body=None):
+        if response_body is None:
+            warnings.warn('ImagesPipeline.convert_image() method called in a deprecated way, '
+                          'method called without response_body argument.',
+                          category=ScrapyDeprecationWarning, stacklevel=1)
```

Comment:
:lipstick: Since the issue is in the calling code, `2` is probably a more useful stack level here. 

```suggestion
                          category=ScrapyDeprecationWarning, stacklevel=2)
```